transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java

```diff
@@ -289,19 +291,23 @@ public void run(IProgressMonitor monitor) {
 
                     if (!saveCamelConfig) {
                         // now update the camel config if we didn't already
-                        CamelContextElement camelContext = camelModel.getCamelContext();
-
-                        // Wizard completed successfully; create the necessary
-                        // config
-                        addCamelContextEndpoint(camelContext, endpoint);
-                        if (sourceFormat != null) {
-                            addDataFormat(camelContext, sourceFormat);
-                        }
-                        if (targetFormat != null) {
-                            addDataFormat(camelContext, targetFormat);
-                        }
-                        // Create the route endpoint
-                        routeEndpoint = new org.fusesource.ide.camel.model.service.core.model.CamelEndpoint("ref:" + endpoint.getId()); //$NON-NLS-1$
+                    	if (camelModel.getRouteContainer() instanceof CamelContextElement) {
+	                        CamelContextElement camelContext = (CamelContextElement)camelModel.getRouteContainer();
+	
+	                        // Wizard completed successfully; create the necessary
+	                        // config
+	                        addCamelContextEndpoint(camelContext, endpoint);
+	                        if (sourceFormat != null) {
+	                            addDataFormat(camelContext, sourceFormat);
+	                        }
+	                        if (targetFormat != null) {
+	                            addDataFormat(camelContext, targetFormat);
+	                        }
+	                        // Create the route endpoint
+	                        routeEndpoint = new org.fusesource.ide.camel.model.service.core.model.CamelEndpoint("ref:" + endpoint.getId()); //$NON-NLS-1$
+                    	} else {
+                    		
```

Comment:
empty else?
remove it or put a comment to explain why we do nothing in this case